File: selectionSort.c
#include <stdio.h>
int main(){
    int liste[] = {1, 45, 0, -1, 69, 420, 0, 9999, 849712, -4};
    int length = sizeof(liste)/sizeof(liste[0]);
    
    for (int counter = 0; counter < length; counter++) {
        // find smallest indice
        int smallestPosition = counter;
        for (int counter2 = counter; counter2 < length; counter2++){
            if (liste[counter2] < liste[smallestPosition]){
                smallestPosition = counter2;
            }
        }
        // swap smallest indice to the beginning
        int cache = liste[counter];
        liste[counter] = liste[smallestPosition];
        liste[smallestPosition] = cache;
    }
    for (int counter = 0; counter < length; counter++){
      printf("%d ", liste[counter]);
    }
    printf("\n");
    return 0;
}

File: selectionSort.java
public static void main(int [] array){
    /**
    * Takes an array of integers and sorts it.
    * @param array The array to sort.
    */
    for (int counter = 0; counter < array.length; counter++) {
        /* find smallest indice.
        * using a separate loop so that we leave out 
        * the already sorted values
        */
        int smallestPosition = counter;
        for (int counter2 = counter; counter2 < array.length; counter2++){
            if (array[counter2] < array[smallestPosition]){
                smallestPosition = counter2;
            }
        }
        // swap smallest indice to the end of the sorted values
        int cache = liste[counter];
        liste[counter] = liste[smallestPosition];
        liste[smallestPosition] = cache;
    }
}
